dbms/src/Flash/tests/gtest_projection_executor.cpp

```diff
@@ -224,5 +244,175 @@ try
 }
 CATCH
 
+TEST_F(ExecutorProjectionTestRunner, MultiFunction)
+try
+{
+    MockAstVec functions = {
+        concat(col("s1"), col("s2")),
+        plusInt(col("s3"), col("s4")),
+        minusInt(col("s3"), col("s4")),
+    };
+    ColumnsWithTypeAndName functions_result = {
+        toNullableVec<String>("concat", {"a1", "a2", "a3a3", {}, {}}),
+        toNullableVec<Int64>("plusInt", {3, {}, 7, {}, 11}),
+        toNullableVec<Int64>("minusInt", {1, {}, 1, {}, 1})};
+    auto test_single_function = [&](size_t index) {
+        auto req = context
+                       .scan("test_db", "test_table2")
+                       .project({functions[index]})
+                       .build(context);
+        executeWithConcurrency(req, {functions_result[index]});
+    };
+    for (size_t i = 0; i < functions.size(); ++i)
+        test_single_function(i);
+
+    auto multi_functions = [&](const MockAstVec & fs) {
+        return context
+            .scan("test_db", "test_table2")
+            .project(fs)
+            .build(context);
+    };
+
+    auto multi_functions_then_agg = [&](const MockAstVec & fs) {
+        return context.scan("test_db", "test_table2")
+            .project(fs)
+            .aggregation({Count(lit(Field(static_cast<UInt64>(1))))}, {})
+            .build(context);
+    };
+
+    MockAstVec fs;
+    ColumnsWithTypeAndName fs_result;
+    for (size_t i = 0; i < functions.size(); ++i)
```

Comment:
I have seen this code pattern several times, maybe can wrap it?

Reply:
`for (size_t i = 0; i < functions.size(); ++i)`?
only appeared twice..
It feels too trivial to wrap it as a method...